ts/component/dropdown.ts

```diff
@@ -44,16 +47,39 @@ export class Dropdown {
       dropdownMenuContainer.classList.remove("ring-2");
       dropdownClickout.classList.toggle("hidden");
     };
-    const invalidInput = () => {
-      dropdownMenuContainer.classList.add(
-        "ring-green-200",
-        "ring-2",
-        "ring-offset-1"
-      );
+    const toggleDropdownButton = () => {
+      // toggle whether or not a labeled button is removed
+      if (displayTemporary) {
+        dropdownButton.classList.toggle("hidden");
+      }
+      toggleMenu();
     };
     dropdownButton.onclick = toggleMenu;
     dropdownButton.innerHTML = makeDisplayText(displayLabel, defaultOption);
-    dropdownClickout.onclick = defaultOption ? toggleMenu : invalidInput;
+    // close dropdown menu by clicking outside of the menu or by hitting Esc
+    dropdownClickout.onclick = toggleDropdownButton;
+    document.addEventListener("keydown", (event) => {
```

Comment:
These listeners will outlive their usefulness and continue to accumulate. Make sure to remove the listener when removing the dropdown